core/src/main/java/org/axonframework/eventhandling/EventProcessor.java

```diff
@@ -59,6 +61,13 @@
      */
     Registration registerInterceptor(MessageHandlerInterceptor<? super EventMessage<?>> interceptor);
 
+    /**
+     * Return the list of registered interceptors for this event processor
+     *
+     * @return the list of interceptors of this event processor
+     */
+    Set<MessageHandlerInterceptor<? super EventMessage<?>>> getInterceptors();
```

Comment:
I went for the `Set` because that is the type of the interceptors in the `AbstractEventProcessor`

```
private final Set<MessageHandlerInterceptor<? super EventMessage<?>>> interceptors = new CopyOnWriteArraySet<>();
```

Why initially a Set was chosen, instead of a List, I am not sure.

Reply:
I will add a default implementation

Reply:
I noticed the implementation in `AbastractEventProcessor`. The question was more related to @smcvb / @abuijze. I'm interested in reasons why it was implemented like that. After we make this method public, it's going to be a bit harder to revert it (if there is a reason for that ofc).